src/test/java/io/jenkins/plugins/folderauth/SerializationTest.java

```diff
@@ -0,0 +1,102 @@
+package io.jenkins.plugins.folderauth;
+
+import com.google.common.collect.ImmutableSet;
+import hudson.model.User;
+import hudson.security.ACL;
+import hudson.security.ACLContext;
+import hudson.util.XStream2;
+import io.jenkins.plugins.folderauth.roles.FolderRole;
+import io.jenkins.plugins.folderauth.roles.GlobalRole;
+import jenkins.model.Jenkins;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static io.jenkins.plugins.folderauth.misc.PermissionWrapper.wrapPermissions;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class SerializationTest {
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Before
+    public void setUp() {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+
+        Set<GlobalRole> globalRoles = new HashSet<>();
+        globalRoles.add(new GlobalRole("admin", wrapPermissions(Jenkins.ADMINISTER),
+                ImmutableSet.of("admin")));
+
+        Set<FolderRole> folderRoles = new HashSet<>();
+        folderRoles.add(new FolderRole("read", wrapPermissions(Jenkins.READ), ImmutableSet.of("user1")));
+
+        j.jenkins.setAuthorizationStrategy(new FolderBasedAuthorizationStrategy(globalRoles, folderRoles));
+    }
+
+    @Test
+    @Issue("JENKINS-58485")
+    public void shouldNotUseConcurrentHashMap$KeySetView() {
+        XStream2 xStream = new XStream2();
+        String xml = xStream.toXML(j.jenkins.getAuthorizationStrategy());
+        assertFalse(xml.contains("ConcurrentHashMap$KeySetView"));
```

Comment:
That is one way to test things, but you are doing too much work, and not really getting that much value out of the test because you are asserting that something you “know” is broken is _not_ done, but you are not asserting that what you are doing instead actually _works_.

Better to replace both test methods with one, and use `RestartableJenkinsRule` to simply assert that after being saved, the plugin’s configuration works as intended after a restart, without specifying what precisely it is saving. (If you do wish to assert compatibility with some _historical_ serial form, use `@LocalData`.)